src/conn_sock.c

```diff
@@ -13,14 +13,46 @@
 #include <sys/un.h>
 #include <sys/stat.h>
 
-static gboolean attach_cb(int fd, G_GNUC_UNUSED GIOCondition condition, G_GNUC_UNUSED gpointer user_data);
-static gboolean conn_sock_cb(int fd, GIOCondition condition, gpointer user_data);
-static gboolean read_conn_sock(struct conn_sock_s *sock);
-static gboolean terminate_conn_sock(struct conn_sock_s *sock);
-void conn_sock_shutdown(struct conn_sock_s *sock, int how);
-static void sock_try_write_to_masterfd_stdin(struct conn_sock_s *sock);
-static gboolean masterfd_write_cb(G_GNUC_UNUSED int fd, G_GNUC_UNUSED GIOCondition condition, G_GNUC_UNUSED gpointer user_data);
-
+static gboolean attach_cb(int fd, G_GNUC_UNUSED GIOCondition condition, gpointer user_data);
+static gboolean rmt_sock_cb(int fd, GIOCondition condition, gpointer user_data);
+static void init_rmt_sock(struct rmt_sock_s *sock, struct rmt_sock_s *src);
+static gboolean read_rmt_sock(struct rmt_sock_s *sock);
+static gboolean terminate_rmt_sock(struct rmt_sock_s *sock);
+static void rmt_sock_shutdown(struct rmt_sock_s *sock, int how);
+static void schedule_lcl_sock_write(struct lcl_sock_s *lcl_sock);
+static void sock_try_write_to_lcl_sock(struct rmt_sock_s *sock);
+static gboolean lcl_sock_write_cb(G_GNUC_UNUSED int fd, G_GNUC_UNUSED GIOCondition condition, G_GNUC_UNUSED gpointer user_data);
+
+static struct lcl_sock_s lcl_masterfd_stdin = {&masterfd_stdin, true, NULL, "container stdin", NULL};
+struct rmt_sock_s rmt_attach_sock = {
```

Comment:
can we have more explaination here on the reasoning behind setting these values and what these structures represent?